javalib/src/main/scala/java/time/Duration.scala

```diff
@@ -0,0 +1,271 @@
+
+  def isZero(): Boolean = seconds == 0 && nanos == 0
+
+  def isNegative(): Boolean = seconds < 0
+
+  def getSeconds(): Long = seconds
+
+  def getNano(): Int = nanos
+
+  def withSeconds(seconds: Long): Duration =
+    new Duration(seconds, nanos)
+
+  def withNanos(nanosOfSecond: Int): Duration =
+    new Duration(seconds, nanosOfSecond)
+
+  def plus(duration: Duration): Duration = {
+    val seconds1 = duration.getSeconds
+    val sumNanos = nanos + duration.getNano
+    if (seconds1 >= 0) {
+      val sumSeconds = Math.addExact(seconds, seconds1)
+      if (sumNanos >= NANOS_IN_SECOND)
+        new Duration(Math.incrementExact(sumSeconds), sumNanos - NANOS_IN_SECOND)
+      else
+        new Duration(sumSeconds, sumNanos)
+    } else {
+      val sumSeconds = Math.addExact(seconds, seconds1 + 1)
+      if (sumNanos >= NANOS_IN_SECOND)
+        new Duration(sumSeconds, sumNanos - NANOS_IN_SECOND)
+      else
+        new Duration(Math.decrementExact(sumSeconds), sumNanos)
+    }
+  }
+
+  def plus(amount: Long, unit: TemporalUnit): Duration = {
+   if (!unit.isDurationEstimated || unit == DAYS)
+      plus(unit.getDuration.multipliedBy(amount))
+    else
+      throw new UnsupportedTemporalTypeException(s"Unit not supported: $unit")
+  }
+
+  def plusDays(days: Long): Duration = plus(days, DAYS)
+
+  def plusHours(hours: Long): Duration = plus(hours, HOURS)
+
+  def plusMinutes(minutes: Long): Duration = plus(minutes, MINUTES)
+
+  def plusSeconds(seconds: Long): Duration = plus(seconds, SECONDS)
+
+  def plusMillis(millis: Long): Duration = plus(millis, MILLIS)
+
+  def plusNanos(nanos: Long): Duration = plus(nanos, NANOS)
+
+  def minus(duration: Duration): Duration = {
+    if (duration == Duration.MIN)
+      plus(Duration.MAX).plusNanos(1)
+    else
+      plus(duration.negated())
+  }
+
+  def minus(amount: Long, unit: TemporalUnit): Duration = {
+    if (!unit.isDurationEstimated || unit == DAYS)
+      minus(unit.getDuration.multipliedBy(amount))
+    else
+      throw new UnsupportedTemporalTypeException(s"Unit not supported: $unit")
+  }
+
+  def minusDays(days: Long): Duration = minus(days, DAYS)
+
+  def minusHours(hours: Long): Duration = minus(hours, HOURS)
+
+  def minusMinutes(minutes: Long): Duration = minus(minutes, MINUTES)
+
+  def minusSeconds(seconds: Long): Duration = minus(seconds, SECONDS)
+
+  def minusMillis(millis: Long): Duration = minus(millis, MILLIS)
+
+  def minusNanos(nanos: Long): Duration = minus(nanos, NANOS)
+
+  def multipliedBy(multiplicand: Long): Duration = {
+    val (prodNanosQuot, prodNanosRem) = {
+      try {
+        val mulNanos = Math.multiplyExact(normalizedNanos, multiplicand)
+        (mulNanos / NANOS_IN_SECOND, (mulNanos % NANOS_IN_SECOND).toInt)
+      } catch {
+        case _: ArithmeticException =>
+          val mulNanos = BigInt(normalizedNanos) * multiplicand
+          ((mulNanos / NANOS_IN_SECOND).toLong, (mulNanos % NANOS_IN_SECOND).toInt)
+      }
+    }
```

Comment:
Wouldn't it be more direct to multiply the `seconds` and `nanos` apart. Then remove the excedent from the `nanos` multiplication and add it to the seconds.

There would be to overflow possible. The on on `seconds`, that is unavoidable and means that `multipliedBy` has overflown. The one on `nanos` must be handled explicitly because part of that could go to the seconds.